File: coupon-login-api/src/main/java/com/kakao/mobility/kmcoupon/interfaces/AuthController.java
package com.kakao.mobility.kmcoupon.interfaces;

import com.kakao.mobility.kmcoupon.application.AuthService;
import com.kakao.mobility.kmcoupon.domain.member.Member;
import com.kakao.mobility.kmcoupon.dto.request.LoginRequest;
import com.kakao.mobility.kmcoupon.dto.response.TokenResponse;
import com.kakao.mobility.kmcoupon.util.JwtTokenUtil;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.Authentication;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;

@RestController
public class AuthController {

    private final AuthService authService;
    private final JwtTokenUtil jwtTokenUtil;

    public AuthController(AuthService authService, JwtTokenUtil jwtTokenUtil) {
        this.authService = authService;
        this.jwtTokenUtil = jwtTokenUtil;
    }

    @PostMapping("/api/v1/auth")
    public ResponseEntity<TokenResponse> signIn(@Valid @RequestBody LoginRequest loginRequest) {
        Authentication authenticatedMember = authService.authenticateMember(loginRequest);
        SecurityContextHolder.getContext().setAuthentication(authenticatedMember);
        String token = jwtTokenUtil.createToken(authenticatedMember);
        return ResponseEntity.ok(new TokenResponse(token));
    }
}

File: coupon-login-api/src/main/java/com/kakao/mobility/kmcoupon/controller/AuthController.kt
package com.kakao.mobility.kmcoupon.controller

import com.kakao.mobility.kmcoupon.service.AuthService
import com.kakao.mobility.kmcoupon.dto.request.LoginRequest
import com.kakao.mobility.kmcoupon.dto.response.TokenResponse
import com.kakao.mobility.kmcoupon.util.JwtTokenUtil
import io.swagger.annotations.Api
import io.swagger.annotations.ApiOperation
import org.springframework.http.ResponseEntity
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import javax.validation.Valid

@RestController
@Api(tags = ["인증 API"])
class AuthController(
    private val authService: AuthService,
    private val jwtTokenUtil: JwtTokenUtil
) {
    @PostMapping("/api/v1/auth")
    @ApiOperation(value = "로그인")
    fun signIn(@Valid @RequestBody loginRequest: LoginRequest): ResponseEntity<TokenResponse> {
        val authenticatedMember = authService.authenticateMember(
            email = loginRequest.email,
            password = loginRequest.password
        )

        SecurityContextHolder.getContext().authentication = authenticatedMember
        val token = jwtTokenUtil.createToken(authenticatedMember)
        return ResponseEntity.ok(TokenResponse(token))
    }
}